borsh-schema-derive-internal/src/attribute_helpers/field/schema.rs

```diff
@@ -0,0 +1,82 @@
+use std::collections::BTreeMap;
+
+use crate::attribute_helpers::{
+    parsing::{meta_get_by_symbol_keys, parse_lit_into_vec},
+    Symbol, DECLARATION, DEFINITIONS, PARAMS, WITH_FUNCS,
+};
+use once_cell::sync::Lazy;
+use syn::{meta::ParseNestedMeta, Ident, Token, Type};
+
+use self::with_funcs::{WithFuncs, WITH_FUNCS_FIELD_PARSE_MAP};
+
+pub mod with_funcs;
+
+pub(crate) enum Variants {
+    Params(Vec<ParameterOverride>),
+    WithFuncs(WithFuncs),
+}
+
+type ParseFn = dyn Fn(Symbol, Symbol, &ParseNestedMeta) -> syn::Result<Variants> + Send + Sync;
+
+pub(crate) static SCHEMA_FIELD_PARSE_MAP: Lazy<BTreeMap<Symbol, Box<ParseFn>>> = Lazy::new(|| {
+    let mut m = BTreeMap::new();
+    // has to be inlined; assigning closure separately doesn't work
```

Comment:
> assigning closure separately doesn't work

What does this mean?

P.S. It would be great to have proper names instead of `f1`, `f2`, `f3`, ...

Reply:
this means that splitting this in 2 lines doesn't work with very verbose errors
```rust
    let func = |attr_name, meta_item_name, meta| {
        parse_lit_into_vec::<ParameterOverride>(attr_name, meta_item_name, meta)
            .map(Variants::Params)
    };
    let f1: Box<ParseFn> = Box::new(func);
```

Reply:
I see. Well, the comment is quite cryptic. Also, having `Box::new(|| ...)` did not bother me at all. I even tried to inline it into `.insert()`, but it becomes too packed, so I am voting against it, and only provide here for reference:

```rust
pub(crate) static SCHEMA_FIELD_PARSE_MAP: Lazy<BTreeMap<Symbol, Box<ParseFn>>> = Lazy::new(|| {
    let mut m = BTreeMap::<Symbol, Box<ParseFn>>::new();

    m.insert(
        PARAMS,
        Box::new(|attr_name, meta_item_name, meta| {
            parse_lit_into_vec::<ParameterOverride>(attr_name, meta_item_name, meta)
                .map(Variants::Params)
        }),
    );

    m.insert(
        WITH_FUNCS,
        Box::new(|_attr_name, _meta_item_name, meta| {
            let map_result =
                meta_get_by_symbol_keys(WITH_FUNCS, meta, &WITH_FUNCS_FIELD_PARSE_MAP)?;
            let with_funcs: WithFuncs = map_result.into();
            if (with_funcs.declaration.is_some() && with_funcs.definitions.is_none())
                || (with_funcs.declaration.is_none() && with_funcs.definitions.is_some())
            {
                return Err(syn::Error::new_spanned(
                    &meta.path,
                    format!(
                        "both `{}` and `{}` have to be specified at the same time",
                        DECLARATION.1, DEFINITIONS.1,
                    ),
                ));
            }
            Ok(Variants::WithFuncs(with_funcs))
        }),
    );

    m
});
```